### ISSUE TITLE
What's the `cookie_value` in `Session` using for ?

### ISSUE THREAD

**Issue by chenaoxd** [http-rs/async-session#26]
Why the `cookie_value` is not cloned or serialized, I can't get the point of doing this. Can someone explain why we should have this field, and make it not cloned or serialized. Thanks a lot. 😂

**Comment by jbr** [comment]
cookie value is what is stored in the cookie, which may be either a unique identifier or the full encrypted value of the cookie, depending on the session store's implementation. https://docs.rs/async-session/3.0.0/async_session/struct.Session.html#cloning-and-serialization It is removed on clone in order to keep from accidentally persisting it in the session, and the only way to retrieve the cookie value is to consume the session, ensuring that this is only done by the framework that owns the "original" (pre-clone) session that was either retrieved from a session store or created for the current request. For datastore-based session stores (everything except the cookie store), we do not store the cookie value anywhere, and the lookup key is a sha256 hash of that cookie value. This ensure that if anyone compromised the session store database, they would not be able to generate a cookie. There is exactly one copy of the cookie value, and it is no longer available anywhere server-side once the set-cookie header has been assigned.

---

Are you writing a framework implementation of async-session, a session store, or using async-session with an existing framework implementation? The cookie value should not be available to framework users.

**Comment by chenaoxd** [comment]
Yeah, I'm implementing my own `postgresql` session store because I'm currently using diesel instead of `sqlx`, so I want to prevent introducing a new db lib. And I'm integrating it into [axum](https://github.com/tokio-rs/axum), I found that `cookie_value` will not be serialized.

I found this because the `cookie_value` will not be set if only deserialized from previously-stored JSON (and by reading the code). Then I think I should use the `set_cookie_value` to set the `cookie_value` by myself after retrieving it from the database? 🤔
